src/components/Synthetics/UserIncentiveDistributionList/ClaimableAmounts.tsx

```diff
@@ -0,0 +1,366 @@
+import { Trans } from "@lingui/macro";
+import { useCallback, useMemo, useState } from "react";
+import { ImSpinner2 } from "react-icons/im";
+import Skeleton from "react-loading-skeleton";
+import { useLocalStorage } from "react-use";
+import useSWR from "swr";
+import { encodeFunctionData } from "viem";
+
+import { getContract } from "config/contracts";
+import { useSettings } from "context/SettingsContext/SettingsContextProvider";
+import { selectChainId } from "context/SyntheticsStateContext/selectors/globalSelectors";
+import { useSelector } from "context/SyntheticsStateContext/utils";
+import useUserClaimableAmounts, { GLP_DISTRIBUTION_ID } from "domain/synthetics/common/useUserClaimableAmounts";
+import { estimateExecutionGasPrice, getExecutionFeeBufferBps } from "domain/synthetics/fees/utils/executionFee";
+import { useTokenBalances } from "domain/synthetics/tokens";
+import { estimateGasLimit } from "lib/gas/estimateGasLimit";
+import { getGasPrice } from "lib/gas/gasPrice";
+import { helperToast } from "lib/helperToast";
+import { formatBalanceAmount, formatUsd } from "lib/numbers";
+import { sendWalletTransaction, TxnEventName } from "lib/transactions";
+import { WalletSigner } from "lib/wallets";
+import { useIsSafeAccount } from "lib/wallets/useIsSmartAccount";
+import useWallet from "lib/wallets/useWallet";
+import ClaimHandlerAbi from "sdk/abis/ClaimHandler.json";
+import { getToken, NATIVE_TOKEN_ADDRESS } from "sdk/configs/tokens";
+
+import { AlertInfoCard } from "components/AlertInfo/AlertInfoCard";
+import Button from "components/Button/Button";
+import Checkbox from "components/Checkbox/Checkbox";
+import { ToastifyDebug } from "components/ToastifyDebug/ToastifyDebug";
+import TooltipWithPortal from "components/Tooltip/TooltipWithPortal";
+
+function getCallData(tokens: string[], account: string, signature: string, distributionId: bigint) {
+  const params = tokens.map((token) => ({
+    token,
+    distributionId,
+    termsSignature: signature as `0x${string}`,
+  }));
+
+  return encodeFunctionData({
+    abi: ClaimHandlerAbi.abi,
+    functionName: "claimFunds",
+    args: [params, account],
+  });
+}
+
+function createClaimAmountsTransaction(data: {
+  tokens: string[];
+  chainId: number;
+  signer: WalletSigner;
+  account: string;
+  signature: string;
+  distributionId: bigint;
+  claimableTokenTitles: Record<string, string>;
+}) {
+  const { tokens, chainId, signer, account, signature, distributionId, claimableTokenTitles } = data;
+  const callData = getCallData(tokens, account, signature, distributionId);
+
+  return sendWalletTransaction({
+    chainId,
+    signer,
+    to: getContract(chainId, "ClaimHandler"),
+    callData,
+    value: 0n,
+    callback: async (event) => {
+      switch (event.event) {
```

Comment:
let's create callback creator hook to simplify the code here?

Also, maybe we want to send some metrics about these transactions?